clients/js/src/embeddings/TransformersEmbeddingFunction.ts

```diff
@@ -0,0 +1,62 @@
+import { IEmbeddingFunction } from "./IEmbeddingFunction";
+
+// Dynamically import module
+let TransformersApi: Promise<any>;
+
+export class TransformersEmbeddingFunction implements IEmbeddingFunction {
+  private pipelinePromise: Promise<any> | null;
+
+  /**
+   * TransformersEmbeddingFunction constructor.
+   * @param options The configuration options.
+   * @param options.model The model to use to calculate embeddings. Defaults to 'Xenova/all-MiniLM-L6-v2', which is an ONNX port of `sentence-transformers/all-MiniLM-L6-v2`.
```

Comment:
can we add a link to all the options here?

Reply:
You mean [these options](https://huggingface.co/docs/transformers.js/main/en/api/utils/hub#module_utils/hub..PretrainedOptions)?

I didn't include all of these since I don't think users would user some of the other options. Which ones do you think I should add? Also, there are some [additional environment options](https://huggingface.co/docs/transformers.js/main/en/api/env#envenv) which might be useful to set. I didn't see other libraries have these options, but I could add it if you think they are useful.

Reply:
@xenova sorry, i should have been more specific :) what models they can use with it. What else beyond `Xenova/all-MiniLM-L6-v2`?

Reply:
Oh gotcha! So, I actually just made a PR for the HF hub so that we can filter by task. Will most likely not be merged this weekend ;). Right now, you can't filter by `task=feature-extraction`, and you can only list the transformers.js models (https://huggingface.co/models?library=transformers.js)

Once merged though, the link will look something like: https://huggingface.co/models?pipeline_tag=feature-extraction&library=transformers.js

Reply:
ok great! ill merge this as-is for now then =D (when tests pass)

Reply:
Awesome! 🚀